Moovies/Moovies/AssemblyProtocol.swift

```diff
@@ -6,29 +6,37 @@ import UIKit
 
 protocol AssemblyProtocol: AnyObject {
     func buildMain(router: RouterProtocol) -> UIViewController
-    func buildDetail(film: Results?, id: Int, router: RouterProtocol) -> UIViewController
+    func buildDetail(film: Movie?, id: Int, router: RouterProtocol) -> UIViewController
 }
 
 final class MoviewModules: AssemblyProtocol {
     func buildMain(router: RouterProtocol) -> UIViewController {
         let movieAPIService = MovieAPIService()
         let view = MooviesViewController()
-        let films = Film(results: [], totalResults: 0, totalPages: 0, page: 0)
-        let presenter = MainPresentor(view: view, model: films, service: movieAPIService, router: router)
+        let dataStorageService = DataStorageService.shared
```

Comment:
```suggestion
        let dataStorageService = DataStorageService
```
.shared никаких синглтонов